Detectors/CPV/reconstruction/src/Clusterer.cxx

```diff
@@ -0,0 +1,422 @@
+        if (isBadChannel(digitN->getAbsId())) { //remove digit
+          digitNEnergy = 0.;
+        }
+        if (digitNEnergy < o2::cpv::CPVSimParams::Instance().mDigitMinEnergy) {
+          continue;
+        }
+
+        // call (digit,digitN) in THAT oder !!!!!
+        Int_t ineb = Geometry::areNeighbours(digitSeedAbsId, digitN->getAbsId());
+        switch (ineb) {
+          case -1: // too early (e.g. previous module), do not look before j at subsequent passes
+            iFirst = j;
+            break;
+          case 0: // not a neighbour
+            break;
+          case 1: // are neighbours
+            clu->addDigit(digitN->getAbsId(), digitNEnergy, digitN->getLabel());
+            iDigitInCluster++;
+            digitsUsed[j - mFirstDigitInEvent] = true;
+            break;
+          case 2: // too far from each other
+          default:
+            break;
+        } // switch
+      }
+    } // loop over cluster
+  }   // energy theshold
+}
+//__________________________________________________________________________
+void Clusterer::makeUnfoldings(gsl::span<const Digit> digits)
+{
+  //Split cluster if several local maxima are found
+
+  std::array<int, mNLMMax> maxAt; // NLMMax:Maximal number of local maxima
+
+  int numberOfNotUnfolded = mClusters.size();
+
+  for (int i = 0; i < numberOfNotUnfolded; i++) { //can not use iterator here as list can expand
+    FullCluster& clu = mClusters[i];
+    if (clu.getNExMax() > -1) { //already unfolded
+      continue;
+    }
+    char nMultipl = clu.getMultiplicity();
+    char nMax = clu.getNumberOfLocalMax(maxAt);
+    if (nMax > 1) {
+      unfoldOneCluster(clu, nMax, maxAt, digits);
+      clu.setEnergy(0); // will be skipped later
+    } else {
+      clu.setNExMax(nMax); // Only one local maximum
+    }
+  }
+}
+//____________________________________________________________________________
+void Clusterer::unfoldOneCluster(FullCluster& iniClu, char nMax, gsl::span<int> digitId, gsl::span<const Digit> digits)
+{
+  // Performs the unfolding of a cluster with nMax overlapping showers
+  // Parameters: iniClu cluster to be unfolded
+  //             nMax number of local maxima found (this is the number of new clusters)
+  //             digitId: index of digits, corresponding to local maxima
+  //             maxAtEnergy: energies of digits, corresponding to local maxima
+
+  // Take initial cluster and calculate local coordinates of digits
+  // To avoid multiple re-calculation of same parameters
+  char mult = iniClu.getMultiplicity();
+  if (meInClusters.capacity() < mult) {
+    meInClusters.reserve(mult);
+    mfij.reserve(mult);
+  }
+
+  const std::vector<FullCluster::CluElement>* cluElist = iniClu.getElementList();
+
+  // Coordinates of centers of clusters
+  std::array<float, mNLMMax> xMax;
+  std::array<float, mNLMMax> zMax;
+  std::array<float, mNLMMax> eMax;
+  std::array<float, mNLMMax> deNew;
+
+  //transient variables
+  std::array<float, mNLMMax> a;
```

Comment:
here also, if possible, use constexpr size.

Reply:
I specially for this reason declared mNLMMax   
`static constexpr short mNLMMax = 10; ///< maximal number of local maxima in cluster`
Or you mean something else?

Reply:
Sorry, did not notice this because according to out coding conventions ``constexpr``'s must be capitalized, could you please change it to ``NLMMax``?